typegraph/node/src/deps/_import.ts

```diff
@@ -0,0 +1,98 @@
+// no-auto-license-header
+
+/*
+The MIT License (MIT)
+
+Copyright (c) 2020 Alexandre Piel
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+export interface Bind {
+  cb?: (file: string) => string;
+}
+
+export const up = 3;
+
+export function caller(this: Bind | any, levelUp = up) {
+  const err = new Error();
+  const stack = err.stack?.split("\n")[levelUp];
+  if (stack) {
+    return getFile.bind(this)(stack);
+  }
+}
+
+export function getFile(this: Bind | any, stack: string) {
+  stack = stack.substr(stack.indexOf("at ") + 3);
+  if (!stack.startsWith("file://")) {
+    stack = stack.substr(stack.lastIndexOf("(") + 1);
+  }
+  const path = stack.split(":");
+  let file = `${path[0]}:${path[1]}`;
+
+  if ((this as Bind)?.cb) {
+    const cb = (this as Bind).cb as any;
+    file = cb(file);
+  }
+  return file;
+}
+
+/*
```

Comment:
once per file should be enough

Reply:
The two sections have different authors... We can merge by listing `Copyright (c) [year] [name]` line I guess.

Reply:
yes